op-proposer/l2_output_submitter.go

```diff
@@ -121,20 +126,33 @@ func NewL2OutputSubmitter(
 ) (*L2OutputSubmitter, error) {
 
 	ctx := context.Background()
+	var l2OutputPrivKey *ecdsa.PrivateKey
+	var err error
 
-	// Parse l2output wallet private key and L2OO contract address.
-	wallet, err := hdwallet.NewFromMnemonic(cfg.Mnemonic)
-	if err != nil {
-		return nil, err
+	if cfg.PrivateKey != "" && cfg.Mnemonic != "" {
```

Comment:
Implementation in the batch-submitter also checks that one of them is enabled
https://github.com/ethereum-optimism/optimism/blob/0e7184b0f09eee193608e78cd0feb1431a2b145a/batch-submitter/config.go#L259

Reply:
Do you want me to change the implementation here to match the one in the BSS?

Reply:
Not super opinionated but could reduce problems at deploy time

Reply:
Will leave as-is for now.